winrt/lib/xaml/BaseControl.h

```diff
@@ -825,7 +840,18 @@ namespace ABI { namespace Microsoft { namespace Graphics { namespace Canvas { na
             return ExceptionBoundary(
                 [&]
                 {
-                    if (--m_loadedCount == 0)
+                    boolean should_unload = false;
+                    if (!WasLoaded())
+                    {
+                        should_unload = true;
+                        auto lock = GetLock();
+                        if (!m_isLoaded) {
+                            should_unload = false;
+                        }
+                        lock.unlock();
```

Comment:
You only need to explicitly call unlock if you want to run something in the same scope but outside the mutex. In this case, unlocking is worthless, since you're going out of scope anyway.